main.go

```diff
@@ -90,6 +90,14 @@ func main() {
 		log.Fatalf("Error reading config: %s", err.Error())
 	}
 
+	hasProfiles, err := k8s.ProfilesEnabled(kubeClient)
+	if err != nil {
+		log.Fatalf("Error getting preferred APIs: %s", err.Error())
+	}
+	if !hasProfiles {
```

Comment:
If this crashes when the CRD isn't present, how does the code get to the handlers for update/deploy? Would this be helpful enough on its own?

Reply:
the profiler CRD can be deleted _before_ startup, which this code catches, or _after_ startup, which the code in the handlers then catches.